### DISCUSSION TITLE
tempPlugin generates grade with wrong version

### DISCUSSION THREAD

**Discussion by bu3alwa** [NativeScript/NativeScript#10295] [Q&A]
So in `platforms/tempPlugin/firebase/build.gradle` I get a gradle version that doesn't exist and I would like to know how to override this so that my build process can stop failing. 

```
buildscript {
    repositories {
        google()
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:7.3.'

        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}
```

I tried adding a buildscript.gradle but it doesn't seem to execute before this process I believe. Is there a way to make it execute before or override it for the specific application?

**Comment by rigor789** [comment]
What version of the cli (`ns -v`) and `@nativescript/android`? That value is usually set based on the runtime version, and perhaps you are using a combination that had the wrong version set?

**Reply by bu3alwa** [reply]
I guess it's not so simple to change that value then? Ok let me give some more details, I have a pipeline which runs latest ns `8.5.3` but it runs into the the issue above. Locally when I run `8.5.3` it compiles without the issue. But locally if I downgrade to `7.2.1` I run into the same issue. Now `@nativescript/android` is at `7.0.1` and I tried to bump it to `8.5.0` but its the same issue. And I did make sure to run `ns clean` when changing anything.

Looking at the source code https://github.com/NativeScript/android/blob/016041e74c28d4d3807df1a2ed6821301c8c0170/package.json#L26
This is the instance where I see the value `7.3.`. What I don't understand is what is the condition in which it uses this value to be backwards compatible vs the other value?

The application is running NS7 and it's not possible to upgrade this as of yet I can only maintain it for now.

**Reply by rigor789** [reply]
The logic is here - should be based on the runtime version, always run latest CLI regardless of your project version, and note you can run `@nativescript/android` latest even against core 7.x.

https://github.com/NativeScript/nativescript-cli/blob/eb918011d6f0be9a8ccb6b569628e3960fd4f8b9/lib/services/android-plugin-build-service.ts#L493-L551

**Reply by bu3alwa** [reply]
Ok thanks, I thought it was the case that I can update them and be fine but your confirmation helps. 

So how do I deal with cases where NS7 has outdated packages and some `build.gradle` files use `compile()`

```
* Where:
Build file '/Users/****/****/****/platforms/tempPlugin/nativescript_google_maps_sdk/build.gradle' line: 232

* What went wrong:
A problem occurred evaluating root project 'nativescript_google_maps_sdk'.
> Could not find method compile() for arguments [com.google.android.gms:play-services-maps:17.0.0] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.
```

**Reply by rigor789** [reply]
Ah that would be a problem if you can't update the plugin itself, in that case you would have to stick with the older runtime I guess.

**Reply by bu3alwa** [reply]
Ok, I'll try to patch it and see how it goes but do you know which `@nativescript/android` version would support `compile()`

**Reply by bu3alwa** [reply]
As I am hacking away at this I am getting this

```
* What went wrong:
A problem occurred evaluating script.
> Could not find method task() for arguments [replaceSettings, settings_e2t2wqgyxqdj916j2uic301y8$_run_closure1@11ee24a9] on settings 'Athena' of type org.gradle.initialization.DefaultSettings.
```

task should still exist on newer gradle versions so I am not sure what it means by that.
